### ISSUE TITLE
There are links in articles that depend on fragile URLs

### ISSUE THREAD

**Issue by leehart** [wtchg-kwiatkowski/observatory-web#292]
E.g.
Home [page] > "Hunting for genes [...]" [heading] > "Read more..." [link] > "Hunting for genes [...]" [page]
See that there is a "chloroquine" link with the target `http://malariagen.net/analytics/pf_resgenes/PF3D7_0709000`

**Comment by benjeffery** [comment]
What's fragile about that link? It isn't a stored state link.

**Comment by leehart** [comment]
Of course, developers and editors might not _expect_ the webapp and table names to change, but since we know that the name of the webapp has changed about 5 times, and the table names, which have recently changed again, do not yet reflect our definitive model for accommodating multiple datasets, it might be prudent to _anticipate_ change.

One position is that this problem (link validity post-change) is the article editor's responsibility, and they should update their links accordingly, if they want to. Personally, I don't expect such a position would guarantee that such broken links within the app will be fixed in a timely manner, or at all. Secondly, I don't think it's fair to expect article editors to go on a search-and-fix mission whenever the webapp or table names happen to change.

One approach is to use something like the `Pretty Links` or `ThirstyAffiliates` plugins for the WordPress CMS, which allow proxy links to be managed within the CMS. For example, the article links to `http://analytics-cms.malariagen.net/foobar/`, which is configured to point somewhere like `http://www.malariagen.net/analytics/foobar`
The advantage would be that all the links that need updating are in one place, and could be updated by whoever changed the URL pattern, rather than the article editors. Those two plugins are already installed and activated in the WordPress CMS, but we're not currently using this approach. I believe a similar approach has been or is being used in the Drupal CMSs that power the MalariaGEN and CGGH websites, so it already has some groupthink weight.

There is also a testing-environment problem, where article editors are expected to check the validity of links that are only available in the staging environment, but they're expected to use links to the production website where the resources don't yet exist. This seems to be traditionally solved using something like environmental variables, e.g. each environment is supplied with a base URL
(one of `http://localhost:8080/analytics`, `http://analytics-staging.malariagen.net/analytics`, `https://www.malariagen.net/analytics`),

I don't have a clear vision about how to solve this yet, although it's cropped up in conversations with stakeholders (such as RA). And I gather we have different opinions about whether there is actually a problem. Any ideas and reassurances welcome!

It might pay to try to head-off this problem sooner rather than later, since I expect the solution could be incorporated into the greatly-anticipated `Writers Guide`, before the problem becomes intractable.

**Comment by benjeffery** [comment]
Can you re-write your example with one of the plugins so we can see how that works?

**Comment by leehart** [comment]
I suspect I might misunderstand what sort of example is needed here, but here's a shot:
E.g.
Home [page] > "Hunting for genes [...]" [heading] > "Read more..." [link] > "Hunting for genes [...]" [page]
1. See that there is a "chloroquine" link with the target http://analytics-cms.malariagen.net/links/chloroquine
2. Click on the "chloroquine" link in 2018 and see that it redirects to http://malariagen.net/analytics/pf_resgenes/PF3D7_0709000
3. Click on the same "chloroquine" link in 2019 and see that it redirects to http://malariagen.net/analytics/pf/genes/PF3D7_0709000/drugs/chloroquine

**Comment by benjeffery** [comment]
Sorry I wasn't clear - I meant in the article mentioned in your example change the URL to use the plugin.

**Comment by leehart** [comment]
I can't do that, even if I wanted to. Richard's article no longer contains that link.
Instead, I can provide a step-by-step guide for a content author/editor such as Richard to do the same thing, which I suppose could be included in the much-needed Article-writer's Guide, _if_ we decide to go this route. (Ideally there would be consistent guidance on how to link to dynamic app resources, from the outset, which we would hope authors/editors would follow, to link-breakage.)
The step-by-step guide example will probably be identical to the user-guide documentation for whichever plugin I happen to use as an example, e.g. https://wordpress.org/plugins/pretty-link/

**Comment by leehart** [comment]
I remember the first post in the CMS contained example links for both plugins, but that post has been deleted now. It looks like the use of the plugins was discussed in Slack, with a couple of screen shots, July 9th 3pm, if that helps. It sounds like you're looking for some real-life examples though. (I'm still not sure why: it seems pretty straight forward: link goes to A, A redirects to B, so later we can change B to C, without having to change the link to A.)
